lib/internal/modules/esm/loader.js

```diff
@@ -52,12 +52,59 @@ let hooksProxy;
  * @typedef {ArrayBuffer|TypedArray|string} ModuleSource
  */
 
+class HooksProxyLoaderDelegate {
```

Comment:
This code is basically the already existing `CustomizedModuleLoader` – it just got moved upwards in this file because it's being used down below.

Reply:
How has it changed other than being moved?

Reply:
Just a bit of duplicate code that was shared with the hooks loader got removed because it lives in the default loader now.